src/System.ComponentModel.Composition/tests/Microsoft/Internal/AssumesTests.cs

```diff
@@ -0,0 +1,252 @@
+            Throws(() =>
+            {
+                Assumes.NotNull<string, string, string>((string)null, "Value", (string)null);
+            });
+        }
+
+        [Fact]
+        public void NotNullOfT1T2T3_NullAsValue2ArgumentAndValue3Argument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.NotNull<string, string, string>("Value", (string)null, (string)null);
+            });
+        }
+
+        [Fact]
+        public void NotNullOfT_ValueAsValueArgument_ShouldNotThrow()
+        {
+            Assumes.NotNull<string>("Value");
+        }
+
+        [Fact]
+        public void NotNullOfT1T2_ValueAsValue1ArgumentAndValue2Argument_ShouldNotThrow()
+        {
+            Assumes.NotNull<string, string>("Value", "Value");
+        }
+
+        [Fact]
+        public void NotNullOfT1T2T3_ValueAsValue1ArgumentAndValue2ArgumentAndValue3Argument_ShouldNotThrow()
+        {
+            Assumes.NotNull<string, string, string>("Value", "Value", "Value");
+        }
+
+        [Fact]
+        public void NotNullOrEmpty_NullAsValueArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.NotNullOrEmpty((string)null);
+            });
+        }
+
+        [Fact]
+        public void NotNullOrEmpty_EmptyAsValueArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.NotNullOrEmpty("");
+            });
+        }
+
+        [Fact]
+        public void NotNullOrEmpty_ValueAsValueArgument_ShouldNotThrow()
+        {
+            var expectations = new List<string>();
+            expectations.Add(" ");
+            expectations.Add("  ");
+            expectations.Add("   ");
+            expectations.Add("Value");
+
+            foreach (var e in expectations)
+            {
+                Assumes.NotNullOrEmpty(e);
+            }
+        }
+
+        [Fact]
+        public void IsTrue1_FalseAsConditionArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.IsTrue(false);
+            });
+        }
+
+        [Fact]
+        public void IsTrue2_FalseAsConditionArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.IsTrue(false, "Message");
+            });
+        }
+
+        [Fact]
+        public void IsTrue1_TrueAsConditionArgument_ShouldNotThrow()
+        {
+            Assumes.IsTrue(true);
+        }
+
+        [Fact]
+        public void IsTrue2_TrueAsConditionArgument_ShouldNotThrow()
+        {
+            Assumes.IsTrue(true, "Message");
+        }
+
+        [Fact]
+        public void NotReachable_ShouldAlwaysThrow()
+        {
+            Throws(() =>
+            {
+                Assumes.NotReachable<object>();
+            });
+        }
+
+        private static void Throws(Action action)
```

Comment:
You're attempting to check for a private/internal exception type, which is actually important information (because " _what_ does it throw?" was my first question).  Possibly rename to `ThrowsNotVisibleException`.